python/TestHarness/TestHarness.py

```diff
@@ -464,143 +524,143 @@ def checkPBSEmulator(self):
-
-                        # Report the current status of JOB_ID
-                        if output_value == 'F':
-                            # F = Finished. Get the exit code reported by qstat
-                            exit_code = int(re.search(r'Exit_status = (-?\d+)', qstat_stdout).group(1))
-                            # Read the stdout file
-                            if os.path.exists(job[2]):
-                                output_file = open(job[2], 'r')
-                                # Not sure I am doing this right: I have to change the TEST_DIR to match the temporary cluster_launcher TEST_DIR location, thus violating the tester.specs...
-                                tester.parameters()['test_dir'] = '/'.join(job[2].split('/')[:-1])
-                                outfile = output_file.read()
-                                output_file.close()
-                                output = tester.processResults(tester.specs['moose_dir'], exit_code, self.options, outfile)
-                                self.testOutputAndFinish(tester, exit_code, outfile)
-                                continue
-                            else:
-                                # I ran into this scenario when the cluster went down, but launched/completed my job :)
-                                reason = 'FAILED (NO STDOUT FILE)'
-                                tester.setStatus(reason, tester.bucket_fail)
-                        elif output_value == 'R':
-                            # Job is currently running
-                            reason = 'RUNNING'
-                        elif output_value == 'E':
-                            # Job is exiting
-                            reason = 'EXITING'
-                        elif output_value == 'Q':
-                            # Job is currently queued
-                            reason = 'QUEUED'
-
-                        if reason != '' and tester.getStatus() != tester.bucket_fail:
-                            tester.setStatus(reason, tester.bucket_pbs)
-                            self.handleTestStatus(tester)
+        # Iterate over all jobs found in the batch file
+        for job_id, job_data in self.pbs_data.iteritems():
+            # retrieve dirpath from pbs
+            dirpath = job_data['test_dir']
+
+            # retrieve 'tests' (-i tests) file location from pbs
+            file = os.path.join(job_data['test_dir'], '../', job_data['input_name'])
+
+            # Populate the input_file_name argument so augmentParameters can format the test_name
+            self.options.input_file_name = job_data['input_name']
+
+            # Enter the directory where the test resides
+            saved_cwd = os.getcwd()
+            sys.path.append(os.path.abspath(dirpath))
+            if os.path.exists(dirpath):
+                os.chdir(dirpath)
+            else:
+                print 'Test/s no longer exists!', job_data['test_dir']
```

Comment:
I was trying to reference either plural or singular here. I suppose I should use the more common syntax:  Test(s)